test/test_dtd.py

```diff
@@ -58,6 +58,36 @@ def test_invalid_inputs(self):
                     self.dtd.validate(parsed), f"{name} validates but shouldn't"
                 )
 
+    def test_backwards_compatibility(self):
+        # the DTD needs to be backwards compatible as long as the major version does not change
+        versions = """
+                ras-dtd-1.0.readalong
+                ras-dtd-1.1.readalong
+            """.strip().split()
+        for name in versions:
+            path = os.path.join(dirname(__file__), "data", name.strip())
+            # DTD is text, XML is binary... okay
+            with open(path, "rb") as infh:
+                parsed = etree.parse(infh)
```

Comment:
We factored out how to correctly and safely load an XML file in `load_xml`. `test_dtd.py` should not use `etree.parse` at all, please add `from readalongs.text.util import load_xml` and use `parse_xml` everywhere we load an XML file in this test file.
`etree.DTD` is appropriate to use, but not `etree.parse`.
This will make the code shorter, too, since `load_xml()` accepts a file name and handles opening it for you.
We also have `parse_xml()` for when the XML comes from a string in memory instead of a file.

Surprisingly, loading XML files the default way can actually be a vulnerability, due to XML bombs (Google it, those are both cool and disturbing!), so it's not just for making the code DRY or shorter, our factored out functions are actually safer.

Reply:
yes please